File: service_invocation/csharp/CheckoutService/Controllers/CheckoutServiceController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using System;
using Microsoft.AspNetCore.Mvc;

namespace CheckoutService.controller
{
    [ApiController]
    [Route("checkout")]
    public class CheckoutServiceController : Controller
    {
        [HttpGet]
        [Route("{orderId}")]
        public ActionResult<string> getCheckout([FromRoute]int orderId)
        {
            Console.WriteLine("Checked out order id : " + orderId);
            return "CID" + orderId;
        }
    }
}

File: state_management/java/CheckoutService/src/main/java/com/service/CheckoutService/controller/CheckoutServiceController.java
package com.service.CheckoutService.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@RestController
@RequestMapping("/")
public class CheckoutServiceController {

    private static final Logger log = LoggerFactory.getLogger(CheckoutServiceController.class);

    @GetMapping(path="/checkout/{orderId}", produces = "application/json")
    public String getCheckout(@PathVariable("orderId") String orderId) {
        log.info("Checked out order id : " + orderId);
        return "CID" + orderId;
    }
}
